tests/unit/components/playback/controls/ButtonWidget.spec.js

```diff
@@ -0,0 +1,326 @@
+    );
+  });
+  test("When the ButtonWidget is mounted, Then it loads the horizontal top-line divider proportion, to the defined width of 490px with padding from edges of the widget length", async () => {
+    const propsData = {
+      button_names: ["Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true],
+      hover_color: ["#BD4932"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_canvas_common_line = wrapper.find(
+      ".canvas__common-horizontal-line"
+    );
+    expect(target_canvas_common_line.attributes().style).toBe("width: 490px;"); // validated if dynamically the value is modified to n-10 px in width as 5px padding is as per mockflow
+  });
+  test("When the ButtonWidget is mounted, Then it loads the widget as there is only one single btn label Save ID, verify that vertical line is not present", async () => {
+    const propsData = {
+      button_names: ["Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true],
+      hover_color: ["#BD4932"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_canvas_vertical_line = wrapper.find(".canvas__vertical-line");
+    expect(target_canvas_vertical_line.exists()).toBe(false);
+  });
+  test("When that ButtonWidget is mounted, Then it loads the button background, with buttons label Cancel/Delete ID/Save ID in between them a vertical line between button labels is rendered", async () => {
+    const propsData = {
+      button_names: ["Cancel", "Delete ID", "Save ID"],
+      enabled_color: "#FFFFFF",
+      disabled_color: "#3F3F3F",
+      is_enabled: [true, true, false],
+      hover_color: ["#BD4932", "#BD4932", "#19ac8a"],
+      button_widget_width: 500,
+      button_widget_height: 50,
+      button_widget_top: 0,
+      button_widget_left: 0,
+    };
+    wrapper = mount(ComponentToTest, {
+      propsData,
+      store,
+      localVue,
+    });
+    const target_vertical_line = wrapper.findAll(".canvas__vertical-line");
+
+    const first_vertical_line = target_vertical_line.at(0);
+    expect(first_vertical_line.attributes().style).toBe(
+      "left: 166.66666666666666px;"
+    );
+    const second_vertical_line = target_vertical_line.at(1);
+    expect(second_vertical_line.attributes().style).toBe(
+      "left: 333.3333333333333px;"
+    );
+  });
+  test("When the ButtonWidget is mounted with Cancel / Save ID buttons and Save is disabled, When User clicks on Cancel, Then an event is emmitted with the Button Index as the contents of the event", async () => {
```

Comment:
the initial setup of the test should be described with "Given" as the first word, not "When".    "When" is used for describing the invocation of the method/function under test.  In this case you are testing the 'click', not the behavior when the component mounts.  mounting the component is just an initial required setup of the test

Reply:
Correct, have fixed the title as "Given"